src/worker-shims.js

```diff
@@ -0,0 +1,27 @@
+import { baseUrl as pageBaseUrl, createBlob } from './common.js';
+
+export class WorkerShim {
+  constructor(aURL, options = {type: 'classic'}) {
+    if (options.type !== 'module') {
+      return new Worker(aURL, options);
+    }
+
+    let es_module_shims_src = new URL('es-module-shims.js', pageBaseUrl).href;
+    const scripts = document.scripts;
+    for (let i = 0, len = scripts.length; i < len; i++) {
+      if (scripts[i].src.includes('es-module-shims.js')) {
+          es_module_shims_src = scripts[i].src;
+
+          break;
+      }
+    }
```

Comment:
Rather than this restriction, could we instead try something like - 

```js
let esModuleShimsSrc;
if (document.currentScript)
  esModuleShimsSrc = document.currentScript.href;
```

Then in the `WorkerShim` constructor, something like - 

```js
if (!esModuleShimsSrc) throw new Error('es-module-shims.js must be loaded with a script tag for WorkerShim support.');
```

note the above would have to be outside of the class constructor and instead in the script initialization.